### DISCUSSION TITLE
log lhs and rhs when cr_assert_eq fails

### DISCUSSION THREAD

**Discussion by xsh005** [Snaipe/Criterion#527] [Q&A] [answered]
In my test files I have a bunch of
```c
cr_assert_eq(get_something(...), expected_value);
```
Currently when it fails, it tells me `the expression ... is false`, it would be more helpful to print out what both sides are, since when I see one I would `cr_log_warn` a message on my own.
Is this functionality already available in criterion? I read the `Assertion reference` and the `Logging messages` section in the docs but I don't see something I described.

**Comment by Snaipe** [comment, marked as answer]
You need to use the [new assertion API](https://criterion.readthedocs.io/en/master/assert.html), which in this case would be `cr_assert(eq(tag, lhs, rhs))`. You can see more examples of this in the samples directory: https://github.com/Snaipe/Criterion/blob/bleeding/samples/asserts.c

**Reply by xsh005** [reply]
Thank you so much!!
